addon.gypi

```diff
@@ -90,10 +90,16 @@
 
     'conditions': [
       [ 'OS=="mac"', {
+        'cflags': [
+          '-fvisibility=hidden'
+        ],
         'defines': [
           '_DARWIN_USE_64_BIT_INODE=1'
         ],
         'xcode_settings': {
+          'OTHER_CFLAGS': [
+            '-fvisibility=hidden'
+          ],
```

Comment:
This can be `GCC_SYMBOLS_PRIVATE_EXTERN`, e.g. https://github.com/nodejs/node/blob/728b155870b69a83b806d8bee67aeb0fd0a9b2dd/deps/uv/uv.gyp#L87 and https://github.com/nodejs/node/blob/f0e460968ef6e24f0882b111abd323f024055675/deps/v8/gypfiles/standalone.gypi#L1060.

https://github.com/nodejs/node-gyp/blob/bb8b294ae76ed16d6633344ff06f6dc5d5ddbe02/gyp/pylib/gyp/xcode_emulation.py#L524-L525